salt/config/__init__.py

```diff
@@ -3315,15 +3315,18 @@ def api_config(path):
     Read in the Salt Master config file and add additional configs that
     need to be stubbed out for salt-api
     '''
-    # Let's grab a copy of salt's master opts
-    opts = client_config(path, defaults=DEFAULT_MASTER_OPTS)
-    # Let's override them with salt-api's required defaults
-    api_opts = DEFAULT_API_OPTS
-    api_opts.update({
+    # Let's grab a copy of salt-api's required defaults
+    opts = DEFAULT_API_OPTS
+
+    # Let's override them with salt's master opts
+    opts.update(client_config(path, defaults=DEFAULT_MASTER_OPTS))
+
+    # Let's set the pidfile and log_file values in opts to api settings
+    opts.update({
         'pidfile': opts.get('api_pidfile', DEFAULT_API_OPTS['api_pidfile']),
         'log_file': opts.get('api_logfile', DEFAULT_API_OPTS['api_logfile']),
```

Comment:
I looked through the history for this file and I do not understand the purpose of setting `pidfile` and `log_file` to the corresponding `api_*` values **at all**. Shot in the dark: I wonder if the daemon that manages netapi modules is expecting those values and should be switched to look for the `api_*` variants.

That said, in order to correctly work with a non-default `root_dir` setting both those values would also need to be added to the `prepend_root_dir` call below.

I think this PR looks good with that change in place (ignoring the `pidfile`/`log_file` unknowns).